framework/src/actions/ActionFactory.C

```diff
@@ -40,15 +42,26 @@ ActionFactory::reg(const std::string & name,
 
 std::shared_ptr<Action>
 ActionFactory::create(const std::string & action,
-                      const std::string & action_name,
+                      const std::string & full_action_name,
                       InputParameters & parameters)
 {
+  std::string action_name = MooseUtils::shortName(full_action_name);
   parameters.addPrivateParam("_moose_app", &_app);
   parameters.addPrivateParam("action_type", action);
   std::pair<ActionFactory::iterator, ActionFactory::iterator> iters;
 
+  if (!(parameters.have_parameter<bool>("isObjectAction") &&
+        parameters.get<bool>("isObjectAction")))
+    parameters.set<std::vector<std::string>>("control_tags")
+        .push_back(MooseUtils::baseName(full_action_name));
+
+  std::string unique_action_name = action + parameters.get<std::string>("task") + full_action_name;
+  // Create the actual parameters object that the object will reference
+  InputParameters & params =
+      _app.getInputParameterWarehouse().addInputParameters(unique_action_name, parameters);
```

Comment:
Can we rename `params` and/or `parameters` to indicate more clearly what their difference is? Maybe `action_params` and `incoming_parser_params` respectively?